File: postgres_driver.py
import psycopg2
import json

def fetch_user_data():
    # Connect to the PostgreSQL database
    conn = psycopg2.connect(
        host="localhost",
        database="postgresql",
        user="nekiarie",
        password="Kiariew@njiiri94"
    )

    # Create a cursor object to interact with the database
    cursor = conn.cursor()

    # Execute the SELECT query
    cursor.execute("SELECT * FROM public.user_table")

    # Fetch all rows from the result set
    rows = cursor.fetchall()

    # Convert the rows to a list of dictionaries
    data = []
    for row in rows:
        d = {
            'user_id': row[0],
            'name': row[1],
            'age': row[2],
            'phone': row[3]
        }
        data.append(d)

    # Convert the data to a JSON string
    json_data = json.dumps(data)

    # Close the cursor and connection
    cursor.close()
    conn.close()

    return json_data

if __name__ == '__main__':
    json_data = fetch_user_data()
    print(json_data)

File: postgres_driver.go
package pgdriver

import (
	"database/sql"
	"encoding/json"

	_ "github.com/lib/pq"
)

type User struct {
	UserID int    `json:"user_id"`
	Name   string `json:"name"`
	Age    int    `json:"age"`
	Phone  string `json:"phone"`
}

type UserTable struct {
	db *sql.DB
}

func NewUserTable(db *sql.DB) *UserTable {
	return &UserTable{
		db: db,
	}
}

func (ut *UserTable) GetUsers() ([]User, error) {
	rows, err := ut.db.Query("SELECT user_id, name, age, phone FROM public.user_table")
	if err != nil {
		return nil, err
	}
	defer rows.Close()

	users := []User{}

	for rows.Next() {
		var user User
		err := rows.Scan(&user.UserID, &user.Name, &user.Age, &user.Phone)
		if err != nil {
			return nil, err
		}
		users = append(users, user)
	}

	return users, nil
}

func (ut *UserTable) GetUsersJSON() ([]byte, error) {
	users, err := ut.GetUsers()
	if err != nil {
		return nil, err
	}

	jsonData, err := json.Marshal(users)
	if err != nil {
		return nil, err
	}

	return jsonData, nil
}
